server.js

```diff
@@ -0,0 +1,130 @@
+const exp = require("constants");
+const cookieParser = require("cookie-parser");
+const express = require("express");
+const { request} = require("http");
+const { type } = require("os");
+const app = express();
+
+const PORT = 8080;
+
+app.use(express.json());
+app.use(express.urlencoded({extended:false}));
+app.use(express.text());
+app.use(cookieParser());
+
+const USERNAME = "Shurahbeel";
+const PASSWORD = "password";
+
+function specialwords(data) {
+    const notallowed = "!@#$%^&*()_+[]{}|;':\",./<>?`~\\-=";
+    for (let i = 0; i < data.length; i++) {
+        if (notallowed.includes(data[i])) {
+            return true;
+        }
+    }
+    return false;
+}
+
+const repo_data = [
+    {
+        "repo_name": "Hackathon prep",
+        "details": "this repo comprises code regarding hackathon 2024",
+        "repo_author": "Shurahbeel Peerzada",
+        "commits": 214,
+        "year_created": 2024,
+    },
+    {
+        "repo_name": "Machine Learning Models",
+        "details": "This repository contains machine learning models and datasets.",
+        "repo_author": "Alexis Jordan",
+        "commits": 150,
+        "year_created": 2023
+    },
+    {
+        "repo_name": "Weather App",
+        "details": "A weather forecasting application using React and Node.js.",
+        "repo_author": "Liam Patterson",
+        "commits": 75,
+        "year_created": 2022
+    },
+    {
+        "repo_name": "E-commerce Backend",
+        "details": "Backend code for an e-commerce platform built with Django.",
+        "repo_author": "Emily Martinez",
+        "commits": 340,
+        "year_created": 2021
+    },
+    {
+        "repo_name": "Portfolio Website",
+        "details": "Personal portfolio website showcasing projects and blogs.",
+        "repo_author": "Shurahbeel Peerzada",
+        "commits": 90,
+        "year_created": 2024
+    },
+    {
+        "repo_name": "Data Visualization Tool",
+        "details": "A tool for visualizing complex data sets using D3.js.",
+        "repo_author": "Nina West",
+        "commits": 125,
+        "year_created": 2023
+    }
+]
+
+app.listen(PORT, () => console.log(`Running on port number: ${PORT}`));
+
+app.post('/login', (req, res) => {
+    const { username, password } = req.body;
+    if (username === USERNAME && password === PASSWORD) {
+        res.cookie('auth', 'loggedin', { httpOnly: true });
+        res.status(201).send("User Successfully Logged in");
+    } else {
+        res.status(404).send("No user found with these credentials");
+    }
+});
+
+app.get('/getallrepos', (req, res) => {
+    if (req.cookies.auth === "loggedin") {
```

Comment:
explore express middlewares and use an auth middleware instead of checking of the cookie for each route.

Reply:
Added the auth middleware wherever necessary